[DOCS] Add deep links to the static banner

pmario opened the pull request:
On static pages at the moment we link to https://tiddlywiki.com instead of the real tiddler.

This PR adds a deep-link to the dynamic tiddler at tw-com.

<img width="1855" height="571" alt="b513f183fc4a608ed2860e5bcb3e3242f70f3c0a" src="https://github.com/user-attachments/assets/6b9fed04-895f-44d6-bc8f-5284e783ee15" />

At the moment the PR is a draft, because I need more testing. 

@Jermolene ... In general. What do you think?

-------

This is related to a SEO discussion at Talk: https://talk.tiddlywiki.org/t/seo-any-expertise-on-getting-better-search-ranking/14223/9

pmario committed:
Add a deep-link to static tiddler pages

pmario committed:
Add deep-links to all static banners

pmario commented:
I did experiment a bit more, with rendering the static page. The landing page now looks like this.

- TOC expanded to level 2, which should give a much better overview
- Page Controls gone
- Search input gone

<img width="1920" height="1040" alt="image" src="https://github.com/user-attachments/assets/7cd90fb3-34fa-4bce-a410-7730fd703a5b" />

@Jermolene 
The next goal is to add a tiddler specific **relative** TOC for every static page. 
What do you think?

pmario committed:
wip add toc to start static page

Bracket-I-I commented:
Probably the best possible change regarding filter documentation 'googling' if it actually does away with just linking to the tiddlywiki frontpage when landing on static pages google spits at one.

Cause holy moly that is always frustrating ...search google for some syntax info, it gives a link to the appropriate page, but it's static, so the 'try me' buttons don't work and there are no edit buttons to see code etc.

And then when ya click on that 'static representation' link it just puts you on the front page and ya gotta search for the same thing again there.

Utterly Infuriating experience. Especially for (goody two shoes cause RTFMing) troglodytes such as myself who hit the documentation all the time.

And because tiddlywiki loves being atomic, you can do this song and dance OFTEN (unless one stays on the tiddlywiki and searches from there) but when I'm in 'docs mode' I tend to just close tabs and the story view is kinda confusing for me for that so I open the documentation tiddlers in new tabs, and all that.

Long story short, if this fixes that then this will probably get rid of a whole lot of (unspoken) rage. It will for sure make me less sore about it all. Cause yeah, that was part of the trifecta of "screw tiddlywiki customization" for me.

pmario commented:
>And then when ya click on that 'static representation' link it just puts you on the front page and ya gotta search for the same thing again there.

Yea, that also was one of the reasons, why I did create this WIP PR. It's the "deep link" part. **But** atm it can only deep-link in the tiddler header section as pointed out in my first screenshot in the OP. 

>...but it's static, so the 'try me' buttons don't work and there are no edit buttons to see code etc.

That will be harder or impossible to fix. This PR hides some of the interactive elements, that do not work anyway. That's suboptimal but better as it is at the moment. 

This PR also adds a new level-parameter to the "simple" TOC so it can be used in the sidebar for more static navigation options. 

There would / may be a possibility to also add an "auto-expand" option for the reveal-widget, so the "try-me" buttons will be rendered expanded for the static view. **But that's not the goal of this PR**

Jermolene commented:
Thanks @Bracket-I-I @pmario. I agree that we should improve the Google search experience as far as we can, and the obvious route to doing so is to make a button that from the users perspective seamlessly turns the static tiddler they are looking at into an interactive TiddlyWiki. We should also make the static banner more prominent (and 
tweaks like ensuring it doesn't get printed).
.

pmario committed:
build staticX is fast

pmario committed:
Add a per tiddler TOC in the sidebar

pmario committed:
Remove redundant tc-reveal divs

pmario committed:
Improve static banner.  Avoid links to system tiddlers

pmario committed:
Remove tc-tiddler-controls from static index page

pmario committed:
Use the Welcome tiddler for static test rendering

pmario committed:
toc links show ... if next level is available

pmario committed:
fix indentation

pmario committed:
make "Back to:" text translateable

pmario committed:
Static page SiteTitle links to tw-com/static. So it works like a "home" link

pmario committed:
Improve simple toc level indicator for static and dynamic TOC

pmario committed:
Merge branch 'master' into static-banner

pmario committed:
ViewTemplate/title fix indentation

pmario committed:
remove test-toc-level. It's part of the toc docs now

pmario committed:
revert temporary changes to tiddlywiki.info

pmario committed:
remove redundant comment from static.tiddler.toc

pmario committed:
Add release note

pmario committed:
externalise CSS from static.html

pmario committed:
Add savefieldcontent build command

Adds a new --savefieldcontent command that decodes a base64-encoded
field of selected tiddlers and writes the bytes as binary files.
Useful for extracting field-stored binary assets (e.g. avatars) into
individual files at build time.

pmario committed:
Externalise images in static build

Static build now extracts image tiddlers and avatar fields to files
alongside the static HTML instead of inlining base64 data URIs.

* Two canonical-uri templates (index vs per-tiddler) compute the
  correct relative src for each render context
* Build target saves images via --savetiddlers, sets _canonical_uri,
  clears the text field, then re-sets _canonical_uri between the
  index render and the per-tiddler render pass
* Community avatar fields are extracted via --savefieldcontent
* Both static templates expose tv-static-avatar-prefix so the
  community-card-display-jpeg-field procedure can render avatar
  links correctly in either context

Reduces static.html from ~1.1 MB to ~219 kB.

pmario committed:
Add Image/ prefix to image tiddler titles

Image tiddlers with no extension in the title (e.g. "Community
Survey 2025") collided with content tiddlers of the same name.
Use a format-agnostic Image/ namespace prefix to avoid collisions
without baking the file extension into references.

Updates all reference points: HelloThumbnail image: fields,
[img[]] wikitext, <$image source=...>, the splash-screen
transclusion, and prose articles. Fixes a few stale .png
references for images that are actually .webp.

pmario committed:
Add tilde to ~TiddlyWiki  to remove the nested A link, which is oiut of HTML spec. Static export for this card is not solved yet

pmario committed:
Merge branch 'master' into static-banner
